indra/preassembler/custom_preassembly.py

```diff
@@ -37,3 +41,22 @@ def agent_name_stmt_type_matches(stmt):
     agents = [agent_name_matches(a) for a in stmt.agent_list()]
     key = str((stmt.__class__.__name__, agents))
     return key
+
+
+@register_pipeline
+def agent_name_stmt_matches(stmt):
+    """Return the normalized agent names."""
+    agents = [agent_name_matches(a) for a in stmt.agent_list()]
```

Comment:
Actually, `agent_name_matches`, as implemented above does some non-standard string processing on the name which I implemented for a specific custom assembly. I would not use it here directly, instead, we could either consider two agents to match if the actual, unmodified `agent.name` matches, or we could do matching based on grounding with name as a fallback.